app/controllers/concerns/sign_in/authentication.rb

```diff
@@ -34,6 +34,22 @@ def load_user(skip_expiration_check: false)
       nil
     end
 
+    def access_token_authenticate
+      access_token.present?
+    rescue Errors::AccessTokenExpiredError => e
+      render json: { errors: e }, status: :forbidden
+    rescue Errors::StandardError => e
+      handle_authenticate_error(e)
+    end
+
+    def access_token_optional_authenticate(skip_expiration_check: false)
+      access_token.present?
+    rescue Errors::AccessTokenExpiredError => e
+      render json: { errors: e }, status: :forbidden unless skip_expiration_check
+    rescue Errors::StandardError
+      nil
+    end
```

Comment:
I think you should be able to combine these and just have the before_action call it with the default param
```ruby
def access_token_authenticate(skip_expiration_check: false)
  access_token.present?
rescue Errors::AccessTokenExpiredError => e
    render json: { errors: e }, status: :forbidden unless skip_expiration_check
rescue Errors::StandardError => e
  handle_authenticate_error(e) unless skip_expiration_check
end
```